pkg/server/storage_api/storage_api_test: TestStatusGetFiles failed

pkg/server/storage_api/storage_api_test.TestStatusGetFiles [failed](https://mesolite.cluster.engflow.com/invocations/default/f2544220-04cf-4e21-a1c5-400f7256276f?testReportRun=3&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc2VydmVyL3N0b3JhZ2VfYXBpOnN0b3JhZ2VfYXBpX3Rlc3Q=) on release-25.4 @ [fae745b39ebade82542387df446d099139cf9cd5](https://github.com/cockroachdb/cockroach/commits/fae745b39ebade82542387df446d099139cf9cd5):


```
=== RUN   TestStatusGetFiles
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestStatusGetFiles1105265807
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected an external process virtual cluster under test; see comment at top of test_server_shim.go for details.
    files_test.go:161: -- test log scope end --
test logs left over in: outputs.zip/logTestStatusGetFiles1105265807
--- FAIL: TestStatusGetFiles (34.47s)
=== RUN   TestStatusGetFiles/cpu
    files_test.go:77: expected 3 files(s), found 4
    files_test.go:83: expected file name cpu3.pprof, found cpuprof.2026-01-20T11_02_14.372.92.pprof
--- FAIL: TestStatusGetFiles/cpu (0.06s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStatusGetFiles.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-58868

--- a/pkg/server/storage_api/files_test.go
+++ b/pkg/server/storage_api/files_test.go
@@ -73,18 +73,24 @@ func TestStatusGetFiles(t *testing.T) {
 				t.Fatal(err)
 			}
 
-			if a, e := len(response.Files), testFilesNo; a != e {
-				t.Errorf("expected %d files(s), found %d", e, a)
+			if a, e := len(response.Files), testFilesNo; a < e {
+				t.Errorf("expected at least %d files(s), found %d", e, a)
 			}
 
-			for i, file := range response.Files {
+			fileMap := make(map[string][]byte, len(response.Files))
+			for _, file := range response.Files {
+				fileMap[file.Name] = file.Contents
+			}
+
+			for i := 0; i < testFilesNo; i++ {
 				expectedFileName := fmt.Sprintf(fileFormatStr, i)
-				if file.Name != expectedFileName {
-					t.Fatalf("expected file name %s, found %s", expectedFileName, file.Name)
-				}
 				expectedFileContents := []byte(fmt.Sprintf("I'm %s profile %d", name, i))
-				if !bytes.Equal(file.Contents, expectedFileContents) {
-					t.Fatalf("expected file contents %s, found %s", expectedFileContents, file.Contents)
+				contents, found := fileMap[expectedFileName]
+				if !found {
+					t.Fatalf("expected to find file %s in response", expectedFileName)
+				}
+				if !bytes.Equal(contents, expectedFileContents) {
+					t.Fatalf("expected file contents %s, found %s", expectedFileContents, contents)
 				}
 			}
 		})

